Remotion/Validation/UnitTests/Attributes/Validation/LengthValidationAttributeTest.cs

```diff
@@ -65,16 +80,75 @@ public void GetPropertyValidator ()
 
       validationMessageStub.Setup(_ => _.ToString()).Returns("Stub Message");
       Assert.That(((LengthValidator)result[0]).ValidationMessage.ToString(), Is.EqualTo("Stub Message"));
+    }
+
+    [Test]
+    public void GetPropertyValidator_WithEqualMinAndMaxValues_ReturnsExactLengthValidator ()
+    {
+      var propertyInformation = PropertyInfoAdapter.Create(typeof(Customer).GetProperty("LastName"));
+      var validationMessageStub = new Mock<ValidationMessage>();
+      _validationMessageFactoryStub
+          .Setup(_ => _.CreateValidationMessageForPropertyValidator(It.IsAny<ExactLengthValidator>(), propertyInformation))
+          .Returns(validationMessageStub.Object);
 
+      var result = _exactLengthValidationAttribute.GetPropertyValidators(propertyInformation, _validationMessageFactoryStub.Object).ToArray();
+
+      Assert.That(result.Length, Is.EqualTo(1));
+      Assert.That(result[0], Is.TypeOf(typeof(ExactLengthValidator)));
```

Comment:
You might want to use ``result.Single()`` here, I don't know if it's preferable. As we already used this implementation in the previous tests however, it's probably going to stay this way.

Reply:
@MichaelKetting Do we want to change it?

Reply:
actually, no, it's fine. 
Before we go to result.Single, we could use 
`Assert.That(result, Has.Exacxly(1).All.TypeOf<ExactLegnthValidator>());`

(not 100% sure about the API, but I think it's close to what is possible)

Or for less compact
```
Assert.That(result.Length, Has.EqualTo(1));
Assert.That(result, Has.All.TypeOf<ExactLegnthValidator>());

Reply:
Also, the `result.Single` gives you much less helpful information incase there's more than one item in the list, so less recommended.

Reply:
Is there a similar implementation of `Is` using `Has` since you used `Has`? If so, what are the differences?

Reply:
`Is` is for single.
`Has` is for collection contraints, i.e. stuff that is applied to each member of the collection.

Reply:
Ok, thank you!